nrm_django/grants/forms.py

```diff
@@ -129,8 +129,20 @@ def __init__(self, *args, **kwargs):
             try:
                 this_cat = Category.objects.get(grant=self.instance)
                 # if we dropped this cat becuase of the distinct() on cat_set, we need to add it back in:
-                if this_cat not in cat_set:
-                    cat_set.append(this_cat)
+                if this_cat not in self.cat_set:
+                    # there is a category in cat_set with this_cat's same
+                    # category_desc. Swap that one out for this one.
+                    cat_ids = [cat.cn for cat in self.cat_set]
+                    replace_me = (
+                        Category.objects.filter(
+                            cn__in=cat_ids, category_desc=this_cat.category_desc
+                        )
+                        .first()
+                        .cn
+                    )
+                    cat_ids[cat_ids.index(replace_me)] = this_cat.cn
+                    # reconstruct the queryset
+                    self.cat_set = Category.objects.filter(cn__in=cat_ids)
                 self.fields["project_category"].initial = this_cat.cn
```

Comment:
Just a thought, and I honestly don't know if it would be more performant, but would it work better if cat_set was used values_list to simply get an array of IDs? that seems like it would simplify replace_me (as well as be faster and more manageable as a simple list rather than a queryset), and then when we need build the real queryset we can still filter with __in like we are now?

it's still a lot of shenanigans for what should just be an FK though

Reply:
I think that `cat_set` has to be a queryset so that it can be specified as the `queryset` parameter of the `ModelChoiceField` on https://github.com/USDAForestService/NRM-Grants-Agreements/blob/development/nrm_django/grants/forms.py#L101 ?

Reply:
Ah, you're right. Another good idea ruined by reality.